sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+                self.add_copy_spec(["/proc/%s/cmdline" % pid,
+                                    "/proc/%s/status" % pid,
+                                    "/proc/%s/mountstats" % pid])
+        self.add_cmd_output(["/bin/ls -ldZ /etc/vdsm",
+                            "/bin/su vdsm -s /bin/sh -c\
+                             '/usr/bin/tree -l /rhev/data-center'",
+                             "/bin/su vdsm -s /bin/sh -c\
+                              '/bin/ls -lR /rhev/data-center'"])
+        self.add_cmd_output("/sbin/lvm vgs -v -o +tags --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_vgs_v_o_tags")
+        self.add_cmd_output("/sbin/lvm lvs -v -o +tags --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_lvs_v_o_tags")
+        self.add_cmd_output("/sbin/lvm pvs -v -o +all --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_pvs_v_o_all")
+        self.add_cmd_output(["/sbin/fdisk -l",
+                             "/usr/bin/iostat",
+                             "/sbin/iscsiadm -m node",
+                             "/sbin/iscsiadm -m session",
+                             "/usr/sbin/nodectl info",
+                             "/usr/bin/abrt-cli list"])
+
+        self.add_cmd_output(['vdsm-client Host getCapabilities',
+                             'vdsm-client Host getStats',
+                             'vdsm-client Host getAllVmStats',
+                             'vdsm-client Host getVMList',
+                             'vdsm-client Host getDeviceList',
+                             'vdsm-client Host hostdevListByCaps',
+                             'vdsm-client Host getAllTasksInfo',
+                             'vdsm-client Host getAllTasksStatuses'])
+
+        res = self.call_ext_prog('vdsm-client Host getConnectedStoragePools')
+        if res['status'] == 0:
+            pools = ast.literal_eval(res['output'])
+            for pool in pools:
+                self.add_cmd_output(
+                    'vdsm-client StoragePool getSpmStatus'
+                    ' storagepoolID={}'.format(pool))
+
+        res = self.call_ext_prog('vdsm-client Host getStorageDomains')
+        if res['status'] == 0:
+            sd_uuids = ast.literal_eval(res['output'])
+            for sd_uuid in sd_uuids:
+                self.add_cmd_output(
+                    'vdsm-tool dump-volume-chains %s' % sd_uuid)
+
+    def _add_vdsm_run_dir(self):
+        """Add everything under /var/run/vdsm except possibly confidential
+        sysprep vfds and sockets"""
+
+        import glob
+
+        for f in glob.glob(
+                "/var/run/vdsm/*"):
+            if not f.endswith('.vfd') and not f.endswith('/isoUploader') \
+                    and not f.endswith('/storage') \
+                    and not stat.S_ISSOCK(os.stat(f).st_mode):
```

Comment:
Sockets are not a problem (they are properly duplicated by the `Archive` classes`), and can in some cases provide useful diagnostic detail (permissions).

Reply:
done